src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java

```diff
@@ -31,22 +35,42 @@ public void promptNextPlayer(){
 
     public void playGame(){
         Scanner keyboardScanner = new Scanner(System.in);
+        
+        Boolean play = true;
+        
+		while(play) {
+			while (board.getWinner() == null){
+				board.printBoard();
+				promptNextPlayer();
+				String line = keyboardScanner.nextLine();
+				String input[] = line.split(",");
+				try {
+					board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
+				} catch (InvalidMoveException e) {
+					System.out.println("Invalid coordinates. Try again");
+					promptNextPlayer();
+				}
+			}
 
-        while (board.getWinner() == null){
-            board.printBoard();
-            promptNextPlayer();
-            String line = keyboardScanner.nextLine();
-            String input[] = line.split(",");
-            try {
-                board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
-            } catch (InvalidMoveException e) {
-                System.out.println("Invalid coordinates. Try again");
-                promptNextPlayer();
-            }
-        }
-
-        board.printBoard();
-        System.out.println("Player " + board.getWinner() + " has won the game!");
+			board.printBoard();
+			if(board.getWinner() == Player.NONE){
+				System.out.println("Game reached a tie!");
+			}else{
+				System.out.println("Player " + board.getWinner() + " has won the game!");
+			}
+			scoreboard.update(board.getWinner());
+			scoreboard.printScore();
+			
+			System.out.println("\nWould you like to play again? (y/n)");
+			String play_again = keyboardScanner.nextLine();
+			if (play_again.equals("n")) {
```

Comment:
**Mandatory Request: ** It would be better if you handle incorrect input. Right now you are only catching for the characters "y" and "n". You should also handle incorrect inputs such as incorrect casing, empty inputs, etc so that your program does not crash.